mirascope/llm/llm_call.py

```diff
@@ -0,0 +1,172 @@
+        from mirascope.core.anthropic import anthropic_call
+
+        return anthropic_call
+    elif provider == "azure":
+        from mirascope.core.azure import azure_call
+
+        return azure_call
+    elif provider == "bedrock":
+        from mirascope.core.bedrock import bedrock_call
+
+        return bedrock_call
+    elif provider == "cohere":
+        from mirascope.core.cohere import cohere_call
+
+        return cohere_call
+    elif provider == "gemini":
+        from mirascope.core.gemini import gemini_call
+
+        return gemini_call
+    elif provider == "groq":
+        from mirascope.core.groq import groq_call
+
+        return groq_call
+    elif provider == "litellm":
+        from mirascope.core.litellm import litellm_call
+
+        return litellm_call
+    elif provider == "mistral":
+        from mirascope.core.mistral import mistral_call
+
+        return mistral_call
+    elif provider == "openai":
+        from mirascope.core.openai import openai_call
+
+        return openai_call
+    elif provider == "vertex":
+        from mirascope.core.vertex import vertex_call
+
+        return vertex_call
+    raise ValueError(f"Unsupported provider: {provider}")
+
+
+def call(
+    model: Models,
+    stream: bool | StreamConfig = False,
+    tools: list[type[BaseTool] | Callable] | None = None,
+    response_model: type[_ResponseModelT] | None = None,
+    output_parser: Callable[[_CallResponseT], _ParsedOutputT]
+    | Callable[[_BaseCallResponseChunkT], _ParsedOutputT]
+    | Callable[[_ResponseModelT], _ParsedOutputT]
+    | None = None,
+    json_mode: bool = False,
+    call_params: CommonCallParams | None = None,
+) -> Callable[
+    [Callable[_P, _R | Awaitable[_R]]],
+    Callable[_P, CallResponse | Stream | Awaitable[CallResponse | Stream]],
+]:
+    provider, model_name = model.split(":", 1)
+    provider_call = _get_provider_call(provider)
+
+    def wrapper(
+        fn: Callable[_P, _R | Awaitable[_R]],
+    ) -> Callable[_P, CallResponse | Stream | Awaitable[CallResponse | Stream]]:
+        decorated = provider_call(
+            model=model_name,
+            stream=stream,
+            tools=tools,
+            response_model=response_model,
+            output_parser=output_parser,
+            json_mode=json_mode,
+            call_params=call_params,
+        )(fn)
+
+        @wraps(decorated)
+        def inner(
+            *args: _P.args, **kwargs: _P.kwargs
+        ) -> CallResponse | Stream | Awaitable[CallResponse | Stream]:
+            result = decorated(*args, **kwargs)
+            if fn_is_async(decorated):
+
+                async def async_wrapper() -> CallResponse | Stream:
+                    final = await result
+                    return _wrap_result(final)
+
+                return async_wrapper()
+            else:
+                return _wrap_result(result)
+
+        return inner
+
+    def _wrap_result(result: BaseCallResponse | Stream) -> CallResponse | Stream:
```

Comment:
nit: define this above `wrapper` since `inner` uses this method (or pull it out into a private utility so we don't define it every time?)